services/autoscaling/src/simcore_service_autoscaling/utils_docker.py

```diff
@@ -46,6 +51,32 @@ async def get_monitored_nodes(node_labels: list[DockerLabelKey]) -> list[Node]:
     return nodes
 
 
+async def remove_monitored_down_nodes(nodes: list[Node]) -> list[Node]:
+    """removes docker nodes that are in the down state"""
+
+    def _check_if_node_is_removable(node: Node) -> bool:
+        if node.Status and node.Status.State:
+            return node.Status.State in [
+                NodeState.down,
+                NodeState.disconnected,
+                NodeState.unknown,
+            ]
+        logger.warning(
+            "%s has no Status/State! This is unexpected and shall be checked",
+            f"{node=}",
+        )
+        # we do not remove a node that has a weird state, let it be done by someone smarter.
+        return False
+
+    nodes_that_need_removal = [n for n in nodes if _check_if_node_is_removable(n)]
```

Comment:
MINOR: do not init docker if nothing to do

```python
nodes_that_need_removal = [n for n in nodes if _check_if_node_is_removable(n)]
if any(nodes_that_need_removal):
   async with aiodocker.Docker() as docker:
       for node in nodes_that_need_removal:
          # ...

return nodes_that_need_removal

Reply:
that is true. but I will init it globally in the app at a later PR.